pkg/asset/manifests/infrastructure.go

```diff
@@ -178,6 +178,7 @@ func (i *Infrastructure) Generate(ctx context.Context, dependencies asset.Parent
 			APIServerInternalIPs: installConfig.Config.Platform.BareMetal.APIVIPs,
 			IngressIPs:           installConfig.Config.Platform.BareMetal.IngressVIPs,
 			LoadBalancer:         installConfig.Config.Platform.BareMetal.LoadBalancer,
+			DNSRecordsType:       installConfig.Config.Platform.BareMetal.DNSRecordsType,
```

Comment:
:bulb: Do we need to set this field in the infrastructure status for platform vsphere, openstack, and nutanix too?